### PR TITLE
fix(userstories kanban): #944 move US at the end of the kanban cell if status, swimlane or project change

### PR DESCRIPTION
**Pull request by bameda**
The main focus of this PR has been to ensure that, when the project, state or swimlane has been changed, the user story is placed at the end (bottom) of the stack of stories in their new cell in the kanban.

In addition, the `UserStory` validator has been modified. Now the` kanban_order` attr cannot be modified with a PATCH / PUT request. It will only be possible with `update_kanban_order_in _bulk` request.

Now to generate `kanban_order` the timestamp will be used in microseconds, to minimize that the kanban_order will be the same in bulk creation.

Some comments have been added, some methods have been reordered and some code that is no longer needed has been deleted.

## How To Validate:

- [ ] Prerequisites
  - [ ] Regenerate data with sample data.
  - [ ] Serve _taiga-front_. 
- [ ] Test 0: Review
  - [ ] Review the code
  - [ ] Review the test code to understand how it works
  - [ ] Pass the tests.
- [ ] Test 1: Change status and/or swimlan from the lightbox kanban modal  
  - [ ] Login as admin 
  - [ ] Go to some projects with kanban, swimlanes and user stories (e. x. project1)
  - [ ] Create a new user story status with the name `test-status-1`
  - [ ] Change status and swimlane of userstories from the edit lightbox and check how they are placed at the end in their new cells.
- [ ] Move [task #944](https://tree.taiga.io/project/taiganext/task/944) to _ready-for-qa_

### CODE REVIEW

File: taiga/base/utils/time.py [hunk 1]
```diff
@@ -21,3 +21,7 @@
 
 def timestamp_ms():
     return int(time.time() * 1000)
+
+
+def timestamp_mics():
```
Comment (yamila-moreno) [inline taiga/base/utils/time.py:26]: ya que la abreviatura oficial no procede, te diría que le pongas porfa un comentario

File: taiga/projects/userstories/api.py [hunk 2]
```diff
@@ -175,52 +195,37 @@ def pre_conditions_on_save(self, obj):
...
-    def _sprint_order_key(self, obj):
-        return "{}-{}-{}".format(obj.project_id, obj.milestone_id, obj.sprint_order)
-
     def pre_save(self, obj):
+        # ## start-hack-reorder ##
+        if obj.id:
+            if self._old_kanban_order_key != self._kanban_order_key(self.object):
+                # The user story is moved to other status, swimlane or project.
+                # It should be at the end of the cell (swimlanes / status).
+                obj.kanban_order = models.UserStory.NEW_KANBAN_ORDER()
```
Comment (yamila-moreno) [inline taiga/projects/userstories/api.py:204]: is this a function/method?

File: taiga/projects/userstories/api.py [hunk 3]
```diff
@@ -277,14 +279,15 @@ def post_save(self, obj, created=False):
...
+        # ## end-hack-rolepoints ##
 
         super().post_save(obj, created)
 
     @transaction.atomic
     def create(self, *args, **kwargs):
         response = super().create(*args, **kwargs)
 
-        # Added comment to the origin (issue)
+        # if US has been prometed from issue, add a comment to the origin (issue)
```
Comment (yamila-moreno) [inline taiga/projects/userstories/api.py:290]: typo "promoted"

File: taiga/projects/userstories/api.py [hunk 4]
```diff
@@ -330,11 +334,13 @@ def update(self, request, *args, **kwargs):
             except Project.DoesNotExist:
                 return response.BadRequest(_("The project doesn't exist"))
 
-        if self.object and self.object.project_id:
-            self._max_order = models.UserStory.objects.filter(
-                project_id=self.object.project_id,
-                status_id=request.DATA.get('status', None)
-            ).aggregate(Max('kanban_order'))['kanban_order__max']
+        # TODO: Remove
```
Comment (yamila-moreno) [inline taiga/projects/userstories/api.py:337]: ?????

File: taiga/projects/userstories/api.py (hunk 3 above)
Comment (bameda) [reply taiga/projects/userstories/api.py]: Done

File: taiga/projects/userstories/api.py (hunk 4 above)
Comment (bameda) [reply taiga/projects/userstories/api.py]: Done

File: taiga/base/utils/time.py (hunk 1 above)
Comment (bameda) [reply taiga/base/utils/time.py:27]: Done

**Pull request merged**